### PR TITLE
Fix missing library issues and make the repo buildable.

### PR DESCRIPTION
**Pull request by dafudeng**
Update googleapi library and remove nonexisting library to make the repo buildable.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -62,9 +62,24 @@ http_archive(
...
-    url = "https://github.com/googleapis/googleapis/archive/68c72c1d1ffff49b7d0019a21e65705b5d9c23c2.tar.gz",
+    sha256 = "0e33a2f12ffb8da6a296d9c30582f01d4ad9e3ecd35aae11b50d4ed3a855946e",
+    strip_prefix = "googleapis-923a4b9a6538087654f02c089cbbaa7c45fd06bd",  # 10/19/2022
+    url = "https://github.com/googleapis/googleapis/archive/923a4b9a6538087654f02c089cbbaa7c45fd06bd.tar.gz",
+)
+
+http_archive(
+    name = "com_google_absl",
+    urls = [
+        "https://github.com/abseil/abseil-cpp/archive/refs/tags/20210324.2.tar.gz",
```
Comment (orionHong) [inline WORKSPACE:73]: Why not use the latest absl release https://github.com/abseil/abseil-cpp/archive/refs/tags/20220623.1.tar.gz?

File: WORKSPACE [hunk 2]
```diff
@@ -62,9 +62,24 @@ http_archive(
...
+    ],
+    patches = [
+        "@//third_party:com_google_absl_f863b622fe13612433fdf43f76547d5edda0c93001.diff"
+    ],
+    patch_args = [
+        "-p1",
+    ],
```
Comment (orionHong) [inline WORKSPACE:80]: I don't understand why we would need to patch a release? Also not too sure where is `@//third_party` referring to?
Also it seems `patches` would conflict with `patch_args` as explained in https://bazel.build/rules/lib/repo/http#http_archive-patches
> patches: list of files that are to be applied as patches after extracting the archive. By default, it uses the Bazel-native patch implementation which doesn't support fuzz match and binary patch, but Bazel will fall back to use patch command line tool if `patch_tool` attribute is specified or there are arguments other than `-p` in `patch_args` attribute.

File: src/service_control_client_impl.cc [hunk 3]
```diff
@@ -172,7 +171,7 @@ void ServiceControlClientImpl::AllocateQuotaFlushCallback(
                        // Cache error response for fail close.
                        AllocateQuotaResponse error_response;
                        QuotaError client_side_error;
-                       *client_side_error.mutable_status() = util::SaveStatusAsRpcStatus(status);
+                       *client_side_error.mutable_status() = google::service_control_client::library_conversion_helper::SaveStatusAsRpcStatus(status);
```
Comment (orionHong) [inline src/service_control_client_impl.cc:174]: nit: maybe we can add `using ::google::service_control_client::library_conversion_helper::SaveStatusAsRpcStatus` and call `SaveStatusAsRpcStatus` directly without the namespace qualifier.

File: WORKSPACE (hunk 1 above)
Comment (dafudeng) [reply WORKSPACE:73]: oops, was experimenting replacing all proto::Status with absl::Status, but it's decided not to go down this path. This import should be dropped.

File: WORKSPACE (hunk 2 above)
Comment (dafudeng) [reply WORKSPACE:80]: Haven't fully looked into it, was playing around with this import, and it should be removed as mentioned.

File: utils/library_conversion_helper.cc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+google::rpc::Status SaveStatusAsRpcStatus(const google::protobuf::util::Status& status) {
+    google::rpc::Status ret;
+    ret.set_code(static_cast<int>(status.code()));
+    ret.set_message(status.message().data(), status.message().size());
+    return ret;
+}
+
+} // namespace library_conversion_helper
+}  // namespace service_control_client
+}  // namespace google
```
Comment (orionHong) [inline utils/library_conversion_helper.cc:32]: nit: missing a space before // namespace library_conversion_helper

File: utils/library_conversion_helper.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#include "google/rpc/status.pb.h"
+
+namespace google {
+namespace service_control_client {
+namespace library_conversion_helper {
+
+// This helper function converts google::protobuf::util::Status to google::rpc::Status.
+google::rpc::Status SaveStatusAsRpcStatus(const google::protobuf::util::Status& status);
+
+} // namespace library_conversion_helper
```
Comment (orionHong) [inline utils/library_conversion_helper.h:28]: ditto

File: src/service_control_client_impl.cc (hunk 3 above)
Comment (dafudeng) [reply src/service_control_client_impl.cc:174]: Then the using statement needs to be dropped upon import, cause internally we don't have using util::SaveStatusAsRpcStatus for the most part.

Comment (orionHong) [reply src/service_control_client_impl.cc:174]: I did a quick search for `using ::util::SaveStatusAsRpcStatus`, and it seems internally we also have quite a few of these in other projects. It doesn't seems like a problem.

File: utils/library_conversion_helper.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/* Copyright 2022 Google Inc. All Rights Reserved.
+
```
Comment (qiwzhang) [inline utils/library_conversion_helper.h:2]: maybe just call it "status.h"

File: utils/library_conversion_helper.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+==============================================================================*/
+
+#ifndef GOOGLE_SERVICE_CONTROL_CLIENT_UTILS_LIBRARY_CONVERSION_HELPER_H_
+#define GOOGLE_SERVICE_CONTROL_CLIENT_UTILS_LIBRARY_CONVERSION_HELPER_H_
+
+#include "google/rpc/status.pb.h"
+
+namespace google {
+namespace service_control_client {
+namespace library_conversion_helper {
```
Comment (qiwzhang) [inline utils/library_conversion_helper.h:23]: name space should be consistent with other files.   just google::service_control_client

File: src/service_control_client_impl.cc (hunk 3 above)
Comment (qiwzhang) [reply src/service_control_client_impl.cc:174]: It seems strange here to use a long namespace prefix.   Can  we use different function name for easy copybara replacement?

Comment (dafudeng) [reply src/service_control_client_impl.cc:174]: Yeah, I think going with orionHong's suggestion makes more sense.
"using ::google::service_control_client::SaveStatusAsRpcStatus" -> "using ::util::SaveStatusAsRpcStatus" upon import.

File: utils/library_conversion_helper.h (hunk 6 above)
Comment (dafudeng) [reply utils/library_conversion_helper.h]: Done.

File: utils/library_conversion_helper.h (hunk 7 above)
Comment (dafudeng) [reply utils/library_conversion_helper.h]: Done.

**Review by qiwzhang — APPROVED** [review]

**Review by TAOXUY — APPROVED** [review]

**Review by nareddyt — APPROVED** [review]

**Pull request merged**